### PR TITLE
ADM-427 [backend][config] - fix pipeline backend in buildKite that is missing two fields

### PR DESCRIPTION
**Pull request by Eason-T**
## Summary
- fix pipeline backend in buildKite that is missing two fields
...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_
add fields in buildKite Backend
**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: backend/src/main/java/heartbeat/controller/pipeline/PipelineController.java [hunk 1]
```diff
@@ -24,8 +27,10 @@ public class PipelineController {
 	private final BuildKiteService buildKiteService;
 
 	@GetMapping("/{pipelineType}")
-	public BuildKiteResponse getBuildKiteInfo(@PathVariable String pipelineType) {
-		return buildKiteService.fetchPipelineInfo();
+	public BuildKiteResponse getBuildKiteInfo(@PathVariable String pipelineType,
+			@RequestHeader("Authorization") @NotBlank(message = "Token must not be blank") String token,
+			@RequestParam String startTime, @RequestParam String endTime) throws NoPermissionException {
```
Comment (bhan-tw) [inline backend/src/main/java/heartbeat/controller/pipeline/PipelineController.java:32]: Is the app handling the `NoPermissionException`?

**Comment by guzhongren** [comment]
Suggestion:
maybe graphql is better than RESTful, you can investigate it, create a card to refactor

File: backend/src/main/java/heartbeat/controller/pipeline/PipelineController.java (hunk 1 above)
Comment (Eason-T) [reply backend/src/main/java/heartbeat/controller/pipeline/PipelineController.java]: done

Comment (bhan-tw) [reply backend/src/main/java/heartbeat/controller/pipeline/PipelineController.java]: Should be http code 401

File: backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java [hunk 2]
```diff
@@ -63,6 +77,17 @@ public BuildKiteResponse fetchPipelineInfo() {
 			log.error("[BuildKite] Failed when call BuildKite", e);
 			throw new RequestFailedException(e);
 		}
+		catch (NoPermissionException e) {
```
Comment (bhan-tw) [inline backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java:80]: Maybe we can use `if` directly, in that case, we can reduce one throw/catch.
Or we can add other handler to handle `NoPermissionException`.

**Review by bhan-tw — APPROVED** [review]
Looks good except the exception handling

**Pull request merged**